tesseract/tesseract_environment/src/kdl/kdl_state_solver.cpp

```diff
@@ -141,6 +141,13 @@ EnvState::Ptr KDLStateSolver::getState(const std::vector<std::string>& joint_nam
   return state;
 }
 
+StateSolver::Ptr KDLStateSolver::clone() const
+{
+  KDLStateSolver::Ptr state_solver = std::make_shared<KDLStateSolver>();
+  state_solver->init(scene_graph_);
```

Comment:
Is this all that really has to be done to clone it? It doesn't need any of the other private members like current_state?

Reply:
Also, should we put a mutex around this as well? Could something else be changing one of these members while you are cloning?